File: pretest programmazione/prova/numero1/sanitization.cpp
#include <stdio.h>
#include <iostream>
#include <vector>
#include <array>

using namespace std;

vector<string> sanitize(vector<string> words, vector<string> banned){
    vector<string> answers;
    for(int i=0; i<words.size(); i++){
        string word = words[i];
        bool isBanned = false;
        for(int j=0; j<banned.size(); j++){
            string ban = banned[j];
            if(word.find(ban) != string::npos){
                isBanned = true;
                break;
            }
        }
        if(!isBanned){
            answers.push_back("SAFE");
        }else
        {
            answers.push_back("BANNED");
        }
    }
    // SCRIVI QUA IL TUO CODICE
    return answers;
}

int main()
{

    freopen("2024-sanitization_sanitization-2_1737988640.txt", "r", stdin); // DECOMMENTA QUA SE VUOI LEGGERE DA FILE
    freopen("output.txt", "w", stdout); // DECOMMENTA QUA SE VUOI SCRIVERE DA FILE

    int N, M;
    vector<string> words, banned, answers;
    cin >> N >> M;

    for(int i = 0; i < M; i++){
        string s;
        cin >> s;
        banned.push_back(s);
    }

    for(int i = 0; i < N; i++){
        string s;
        cin >> s;
        words.push_back(s);
    }

    answers = sanitize(words, banned);

    for(auto s : answers){
        cout << s << endl;
    }

    return 0;
}

File: pretest programmazione/prova/numero1/sanitization.py
#!/bin/env python3

import sys

# Se vuoi leggere/scrivere da file decommenta qua
# fin = open("input.txt", "r")  # File di input fornito dalla piattaforma
# fout = open("output.txt", "w")  # File di output fornito dalla piattaforma

# Se vuoi leggere/scrivere da linea di comando decommenta qua
fin = sys.stdin  # File di input fornito dalla piattaforma
fout = sys.stdout  # File di output fornito dalla piattaforma


def sanitize(words, banned):
    # SCRIVI QUA IL TUO CODICE
    return []

N, M = map(int, fin.readline().strip().split())

words = []
banned = []

for _ in range(M):
    banned.append(fin.readline().strip())

for _ in range(N):
    words.append(fin.readline().strip())

ans = sanitize(words, banned)

for a in ans:
    print(a)